core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -526,43 +551,46 @@ private[kafka] class Processor(val id: Int,
     try selector.poll(300)
     catch {
       case e @ (_: IllegalStateException | _: IOException) =>
-        error(s"Closing processor $id due to illegal state or IO exception")
-        swallow(closeAll())
-        shutdownComplete()
+        error(s"Processor $id poll failed due to illegal state or IO exception")
         throw e
```

Comment:
Do we get any benefit from rethrowing this?

Reply:
I was in two minds about this and thought it was better to leave it as-is. But on second thought, since `clear()` happens right at the start of `poll()`, it may be better to not re-throw and process anything already completed in the poll. Updated.

Reply:
Yes, I think that seems consistent with the general approach in the rest of the code. We should file a JIRA about thinking whether we should consider some exceptions fatal. In that case, we'd want to shutdown the processor altogether (and not just the Selector) as you said.